modules/carto/src/basemap.ts

```diff
@@ -1,4 +1,91 @@
-const baseUrl = 'https://basemaps.cartocdn.com/gl/{basemap}-gl-style/style.json';
+export const cartoBasemapsBaseUrl = 'https://basemaps.cartocdn.com/gl';
+
+const baseUrl = `${cartoBasemapsBaseUrl}/{basemap}-gl-style/style.json`;
+
+export const cartoStyleLayerGroups = [
+  {
+    slug: 'label',
+    filter: ({id}: {id: string}) =>
+      Boolean(
+        id.match(/(?=(label|_label|place-|place_|poi-|poi_|watername_|roadname_|housenumber))/)
+      ),
+    defaultVisibility: true
+  },
+  {
+    slug: 'road',
+    filter: ({id}: {id: string}) =>
+      Boolean(id.match(/(?=(road|railway|tunnel|street|bridge))(?!.*label)/)),
+    defaultVisibility: true
+  },
+  {
+    slug: 'border',
+    filter: ({id}: {id: string}) => Boolean(id.match(/border|boundaries|boundary_/)),
+    defaultVisibility: false
+  },
+  {
+    slug: 'building',
+    filter: ({id}: {id: string}) => Boolean(id.match(/building/)),
+    defaultVisibility: true
+  },
+  {
+    slug: 'water',
+    filter: ({id}: {id: string}) => Boolean(id.match(/(?=(water|stream|ferry))/)),
+    defaultVisibility: true
+  },
+  {
+    slug: 'land',
+    filter: ({id}: {id: string}) =>
+      Boolean(id.match(/(?=(parks|landcover|industrial|sand|hillshade|park_))/)),
+    defaultVisibility: true
+  }
+];
+
+export function applyCartoLayerGroupFilters(style, visibleLayerGroups: Record<string, boolean>) {
+  if (!Array.isArray(style?.layers)) {
+    return style;
+  }
+
+  const visibleFilters = cartoStyleLayerGroups
+    .filter(lg => visibleLayerGroups[lg.slug])
+    .map(lg => lg.filter);
+
+  const filteredLayers = style.layers.filter(layer => visibleFilters.some(match => match(layer)));
+
+  return {
+    ...style,
+    layers: filteredLayers
+  };
+}
+
+function someLayerGroupsDisabled(visibleLayerGroups?: Record<string, boolean>) {
+  return visibleLayerGroups && Object.values(visibleLayerGroups).every(Boolean) === false;
+}
+
+export async function getCartoBasemapStyle({
+  styleType,
+  visibleLayerGroups
+}: {
+  styleType: string;
+  visibleLayerGroups?: Record<string, boolean>;
+}) {
+  /* global fetch, console */
+  const styleUrl = `${cartoBasemapsBaseUrl}/${styleType}-gl-style/style.json`;
+  let style = styleUrl;
+
+  if (visibleLayerGroups && someLayerGroupsDisabled(visibleLayerGroups)) {
+    try {
+      const originalStyle = await fetch(styleUrl, {
```

Comment:
Rather than using `fetch` directly, use `requestWithParameters` if possible along with an `APIErrorContext`

Reply:
`requestWithParameters` does lots of things, like passing `?v=` (deck version), falling back to `POST` not allowing setting `mode`.

I guess that using just `fetch` is simpler than artificially extending `requestWithParameters` to become another fetch-like univeral API.

(added usage of errorContext though)

Reply:
The other major thing requestWithParameters provides is caching of repeated requests, but in all the current examples this doesn't matter and we only call fetchMap once, so I think it's optional here. 👍🏻

Reply:
Well, we do have browser cache already. If we really want to use current cache, i would factor-it-out as separate "fetchWithCache" (and only with cache), but without carto-specific features (?v=..., fallback to POST ...),

Reply:
And when trying to see how `requestWithParameters` can be adapted/simplified i've found that it's cache is not-configurable and doesn't have any means to clear it - i think we should improve it.

Anyway i guess that in current state of afair it's better to rely on browser cache.